CRM/Core/Smarty.php

```diff
@@ -340,4 +342,18 @@ private function getLocale() {
     return 'en_US';
   }
 
+  /**
+   * Get the compile_check value.
+   *
+   * @return bool
+   */
+  private function isSmartyCompiled() {
+    // check for define in civicrm.settings.php as FALSE, otherwise returns TRUE
+    if (defined('CIVICRM_TEMPLATE_COMPILE_CHECK') && (CIVICRM_TEMPLATE_COMPILE_CHECK === FALSE)) {
+      return FALSE;
```

Comment:
@sluc23 question could we if we have defined this could we just return the value of 
`CIVICRM_TEMPLATE_COMPILE_CHECK`